### PR TITLE
Scale listeners for scale begin and scale end

### PR DESCRIPTION
**Pull request by harisk92**
Added:
onScaleBegin callback when scale starts
onScaleEnd callback when scale ends

### CODE REVIEW

File: photoview/src/main/java/com/github/chrisbanes/photoview/PhotoViewAttacher.java [hunk 1]
```diff
@@ -476,7 +491,9 @@ public void setScale(float scale, float focalX, float focalY,
             throw new IllegalArgumentException("Scale must be within the range of minScale and maxScale");
         }
 
+
```
Comment (fawaad) [inline photoview/src/main/java/com/github/chrisbanes/photoview/PhotoViewAttacher.java]: Shouldn't add blank lines when it doesn't add clarity

File: photoview/src/main/java/com/github/chrisbanes/photoview/PhotoViewAttacher.java [hunk 2]
```diff
@@ -236,6 +250,7 @@ public boolean onDoubleTap(MotionEvent ev) {
                     float x = ev.getX();
                     float y = ev.getY();
 
+
```
Comment (fawaad) [inline photoview/src/main/java/com/github/chrisbanes/photoview/PhotoViewAttacher.java]: Shouldn't add blank lines when it doesn't add clarity

**Comment by fawaad** [comment]
This would be a useful PR to integrate into this library.

**Comment by wching** [comment]
@fawaad hello! Is the spacing the only thing that needs to get fixed on this PR? I would gladly create a new one with those fixes (and of course all the credit will be for @harisk92). Maybe @harisk92 is a bit busy atm and I will glady help him out if that's the case. I think this feature will be super useful :)

**Comment by harisk92** [comment]
@fawaad I removed blank lines. @wching @AdamMc331 sorry I was bit lost in space and time didn't had a chance to update PR but now I did.

File: photoview/src/main/java/com/github/chrisbanes/photoview/PhotoViewAttacher.java [hunk 3]
```diff
@@ -475,8 +489,9 @@ public void setScale(float scale, float focalX, float focalY,
         if (scale < mMinScale || scale > mMaxScale) {
             throw new IllegalArgumentException("Scale must be within the range of minScale and maxScale");
         }
-
+
```
Comment (fawaad) [inline photoview/src/main/java/com/github/chrisbanes/photoview/PhotoViewAttacher.java]: Can you remove this change

**Comment by wching** [comment]
@harisk92 no worries! We all have those :) thanks for your effort on updating the PR! 🙇

**Comment by wching** [comment]
Any updates on this? :) sorry to be such a pain in the butt

**Comment by Jawnnypoo** [comment]
Yep, the reason not to merge is that there are unneeded additions, such as the gradle wrapper, as well as the fact that this PR is not backwards compatible, and would break code for anyone using a scale change listener on the current version who updates.

**Comment by wching** [comment]
Understood, thanks @Jawnnypoo.

**Pull request closed without merging**